azurelinuxagent/common/utils/restutil.py

```diff
@@ -364,6 +373,7 @@ def http_request(method,
                  max_retry=None,
                  retry_codes=None,
                  retry_delay=DELAY_IN_SECONDS,
```

Comment:
What's the difference between retry_delay and telemetry_throttle_delay?

Reply:
retly_delay used for non-telemetry calls for both retry errors and throttle errors

but telemetry_throttle_delay used telemetry calls throttle errors

RETRY_CODES = [
    httpclient.RESET_CONTENT,
    httpclient.PARTIAL_CONTENT,
    httpclient.FORBIDDEN,
    httpclient.INTERNAL_SERVER_ERROR,
    httpclient.NOT_IMPLEMENTED,
    httpclient.BAD_GATEWAY,
    httpclient.SERVICE_UNAVAILABLE,
    httpclient.GATEWAY_TIMEOUT,
    httpclient.INSUFFICIENT_STORAGE,
    429,  # Request Rate Limit Exceeded
]

THROTTLE_CODES = [
    httpclient.FORBIDDEN,
    httpclient.SERVICE_UNAVAILABLE,
    429, # Request Rate Limit Exceeded
]

Reply:
I see, so for errors in retry_codes, we will use the retry_delay (for non-telemetry and telemetry calls)

Instead of adding telemetry_throttled_delay parameter, would it make sense to add throttle_delay parameter?
default value for throttle_delay parameter can be same as retry_delay. We can override throttle_delay default only for telemetry calls

Reply:
+1